app/controllers/wikis_controller.rb

```diff
@@ -42,11 +48,19 @@ def destroy
   private
 
   def wikis
-    @wikis ||= Wiki.all
+    if current_user.premium? || current_user.admin?
+      @wikis ||= Wiki.all
+    else
+      @wikis ||= Wiki.where(private: nil)
+    end
   end
 
   def wiki
     @wiki ||= Wiki.find(params[:id])
+    # unless @wiki.readable_by?(current_user)
+    #   redirect_to wikis_path, alert: 'Update to Premium in order to view private wikis.'
+    # end
+    #raise ActiveRecord::RecordNotFound unless @wiki.readable_by?(current_user)
```

Comment:
You could uncomment this line to ensure that the current user has permission to access this object. You will need to ensure the last line of this method is `@wiki` though, so that the code that depends on it gets the correct object.